docs/deployment/overview.md

```diff
@@ -1,47 +1,301 @@
+| Command | What it does | Outputs | Parameter state | Requires integration capability |
+|---------|--------------|---------|-----------------|---------------------------------|
+| `aspire publish` | Transforms the application model into integration-specific assets (Compose files, manifests, specifications, etc.). | Intermediate artifacts (not directly production-final). | Unresolved (placeholders, e.g. `${VAR}` or similar). | Publish support |
+| `aspire deploy` | Runs a deployment using one or more integrations (build, parameter resolution, apply). | Real resources / applied changes. | Resolved. | Deploy support |
 
-### Azure Container Apps
+If an integration does not implement deploy functionality, `aspire deploy` will not deploy that target (it may warn or no-op for it).
 
-.NET Aspire projects are designed to run in containerized environments. Azure Container Apps is a fully managed environment that enables you to run microservices and containerized applications on a serverless platform. The [Azure Container Apps](azure/aca-deployment.md) topic describes how to deploy Aspire apps to ACA using the Aspire CLI or Azure Developer CLI (azd).
+When you run `aspire deploy` without any integrations that support deployment, you'll see this error:
 
-### Use Application Insights for .NET Aspire telemetry
+```Output
+FAILED: Analyzing the distributed application model for publishing and deployment capabilities.
+           No resources in the distributed application model support deployment.
+```
 
-.NET Aspire projects are designed to emit telemetry using OpenTelemetry which uses a provider model. .NET Aspire projects can direct their telemetry to Azure Monitor / Application Insights using the Azure Monitor telemetry distro. For more information, see [Use Application Insights for .NET Aspire telemetry](azure/application-insights.md) for step-by-step instructions.
+Similarly, when you run `aspire publish` without any integrations that support publishing, you'll see:
 
-## Deploy to Kubernetes
+```Output
+Analyzing the distributed application model for publishing and deployment capabilities. 00:00:00
+           No resources in the distributed application model support publishing.
+```
 
-Kubernetes is a popular container orchestration platform that can run .NET Aspire projects. To deploy .NET Aspire projects to Kubernetes clusters, you need to map the .NET Aspire JSON manifest to a Kubernetes YAML manifest file. There are two ways to do this: by using the Aspir8 project, or by manually creating Kubernetes manifests.
+These messages indicate that you need to add deployment integrations to your AppHost project. Deployment integrations are NuGet packages (like `Aspire.Hosting.Docker`, `Aspire.Hosting.Kubernetes`, or `Aspire.Hosting.Azure`) that provide the publishing and deployment capabilities for specific target platforms.
```

Comment:
I'd prefer to not include a new integration category, these are simply hosting integrations. I'd shy away from calling them deployment integrations.